### PR TITLE
feat: [IDP-368]: Plugin request flow

### PR DESCRIPTION
**Pull request by sathish-soundarapandian**
## Describe your changes
Plugin request flow

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [x] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger utAll`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/46415)

### CODE REVIEW

**Comment by sathish-soundarapandian** [comment]
trigger ss

File: idp-service/contracts/openapi/v1/openapi.yaml [hunk 1]
```diff
@@ -1723,6 +1788,12 @@ components:
         application/json:
           schema:
             $ref: '#/components/schemas/HarnessEntitiesCountResponse'
+    PluginRequestResponseList:
```
Comment (mankrit-singh) [inline idp-service/contracts/openapi/v1/openapi.yaml:1791]: please add the Pagination headers

Comment (sathish-soundarapandian) [reply idp-service/contracts/openapi/v1/openapi.yaml:1808]: Done.

**Review by sandeepa-kv-jois — APPROVED** [review]

**Review by mankrit-singh — COMMENTED** [review]
> Reviewed 3 of 12 files at r1, 12 of 12 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/46415)* status: 15 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained.

**Review by mankrit-singh — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/46415)* status: 15 file(s) to review.

**Review by vikyathharekal — APPROVED** [review]
> Reviewed 3 of 12 files at r1, 12 of 12 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/46415)* status: :shipit: complete! ✓

___
*[`idp-service/contracts/openapi/v1/openapi.yaml` line 651 at r2](https://reviewable.io/reviews/harness/harness-core/46415#-NTHktgJ9ByIPMHpy36g:-NTHktgJ9ByIPMHpy36h:b5ncwn7) ([raw file](https://github.com/harness/harness-core/blob/be10cbcc07a37e544aceb37bf22c27f9ee8a01f3/idp-service/contracts/openapi/v1/openapi.yaml#L651)):*
> ```YAML
>     get:
>       summary: Get all plugin requests for an account
>       operationId: get-plugin-request
> ```

nit: how about `get-plugin-requests`?



_Code quote:_
```YAML
get-plugin-request
```
___
*[`idp-service/src/main/java/io/harness/idp/plugin/mappers/PluginRequestMapper.java` line 22 at r2](https://reviewable.io/reviews/harness/harness-core/46415#-NTHm_YC-rDP6YxsgD4N:-NTHm_YC-rDP6YxsgD4O:b-40h6uz) ([raw file](https://github.com/harness/harness-core/blob/be10cbcc07a37e544aceb37bf22c27f9ee8a01f3/idp-service/src/main/java/io/harness/idp/plugin/mappers/PluginRequestMapper.java#L22)):*
> ```Java
>   public PluginRequestEntity fromDTO(String accountIdentifier, RequestPlugin pluginRequest) {
>     PluginRequestEntity pluginRequestEntity = new PluginRequestEntity();
>     pluginRequestEntity.setAccountIdentifier(accountIdentifier);
> ```

We should consider using ModelMapper in future



_Code quote:_
```Java
luginRequestMapper
```

**Comment by vikyathharekal** [comment]
trigger review

**Pull request merged**